src/app/(admin)/(auth)/register/page.tsx

```diff
@@ -0,0 +1,131 @@
+'use client'
+
+import Link from 'next/link'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useForm } from 'react-hook-form'
+import { z } from 'zod'
+
+import { Button } from '@/components/ui/button'
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import { Separator } from '@/components/ui/separator'
+
+const formSchema = z.object({
+  username: z.string().min(2, {
+    message: 'Username must be at least 2 characters.',
+  }),
+})
```

Comment:
Your schema is not what the form is. Please update the schema and include all the fields.

Should be like this:

```tsx
const formSchema = z.object({
  firstName: z.string().min(2, {
    message: 'First name must be at least 2 characters.',
  }),
  lastName: z.string().min(2, {
    message: 'Last name must be at least 2 characters.',
  }),
  email: z
    .string()
    .email({ message: 'Invalid email address.' })
    .nonempty({ message: 'Email is required.' }),
  password: z
    .string()
    .min(8, { message: 'Password must be at least 8 characters.' }),
  confirmPassword: z
    .string()
    .min(8, { message: 'Confirm Password must be at least 8 characters.' })
    .refine((val, ctx) => val === ctx.parent.password, {
      message: 'Passwords must match.',
    }),
})
```